sound/soc/sof/intel/hda.h

```diff
@@ -409,6 +411,8 @@ struct sof_intel_hda_dev {
 	int irq;
 
 	/* PM related */
+	u32 dsp_power_state;
+	struct mutex ps_mutex;	/* protects dsp_power_state change */
```

Comment:
I am also not able to fully appreciate the need for this mutex, at least just yet while reviewing only the second patch of the series. We only set the power states during suspend/resume and they can never overlap

Reply:
@keyonjie that was my other comment, why do we need to do it in PCM open/close, can we not do it just when we send the IPC?

Reply:
@keyonjie This is not a simple lock. In the comment this indicates the lock protects "dsp_power_state", but reading the code, the lock actually covers DSP power state as well. E.g. if you change the DSP power state, you need to hold this lock. Do we really need, or is a lock protecting just the dsp_power_state enough (and if yes, do we need a lock at all)?